sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/FailFastRntbdRequestRecord.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.directconnectivity.rntbd;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.RequestTimeoutException;
+import com.azure.cosmos.implementation.http.HttpHeaders;
+import io.netty.util.Timeout;
+import io.netty.util.TimerTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.SocketAddress;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.azure.cosmos.implementation.guava27.Strings.lenientFormat;
+
+public class FailFastRntbdRequestRecord extends RntbdRequestRecord {
+    private static final Logger logger = LoggerFactory.getLogger(RntbdRequestRecord.class);
+
+    private FailFastRntbdRequestRecord(final RntbdRequestArgs args) {
+        super(args);
+    }
+
+    @Override
+    public Timeout newTimeout(final TimerTask task) {
+        throw new IllegalArgumentException("newTimeout must never be called for fail fast records.");
+    }
+
+    public static FailFastRntbdRequestRecord createAndFailFast(
+        RntbdRequestArgs args,
+        long concurrentRequestsSnapshot,
+        AtomicInteger concurrentRequests,
+        RntbdMetrics metrics,
+        SocketAddress remoteAddress) {
+
+        FailFastRntbdRequestRecord failFastRecord = new FailFastRntbdRequestRecord(args);
+
+        logger.debug(
+            "\n  [{}]\n  {}\n  created FailFastRntbdRequestRecord {} ",
+            failFastRecord,
+            args,
+            concurrentRequests);
+
+        final String reason = lenientFormat(
+            "Failed due to too many (%s) concurrent requests.",
+            concurrentRequestsSnapshot);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(HttpConstants.HttpHeaders.ACTIVITY_ID, failFastRecord.activityId().toString());
```

Comment:
ActivityId for non-service interaction is mis-leading. 
How about not have it?

Reply:
I have replicated here what happens for client-generated GoneException. I would rather not change it - because that would mean I need to check everywhere whether there are assumptions that the ACTIVITY_ID header exists everywhere. Don't want to take that risk right now (and leaving it in GoneException but not here sees worse than whe apporach in the PR). Leaving as is - please reactivate if you disagree.